File: src/main/java/architecture/goldenboughs_lib/client/model/armor/ModGeoArmorModel.java
package architecture.goldenboughs_lib.client.model.armor;

import architecture.goldenboughs_lib.client.model.GeoModelExpand;
import architecture.goldenboughs_lib.core.GoldenBoughsLib;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.NotNull;
import software.bernie.geckolib.animatable.GeoAnimatable;

public class ModGeoArmorModel<T extends GeoAnimatable> extends GeoModelExpand<T> {

	public static final ResourceLocation DEFAULT_MODEL = model(GoldenBoughsLib.modRl("armor/default")).withSuffix(MODEL_SUFFIX);

	public ModGeoArmorModel(ResourceLocation pathName) {
		this(pathName, pathName, pathName);
	}

	public ModGeoArmorModel(ResourceLocation modelPath, ResourceLocation textureName, ResourceLocation animationsName) {
		super(model(getPath(modelPath)), texturePath(getPath(textureName)), animationsPath(getPath(animationsName)));
	}

	public static ResourceLocation getPath(ResourceLocation name) {
		return name.withPrefix("armor/");
	}

	@NotNull
	@Override
	protected ResourceLocation getDefaultModelResource() {
		return DEFAULT_MODEL;
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/client/model/armor/ModGeoArmorModel.kt
package architecture.goldenboughs_lib.client.model.armor

import architecture.goldenboughs_lib.client.model.GeoModelExpand
import architecture.goldenboughs_lib.core.Lib
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.animatable.GeoAnimatable
import architecture.goldenboughs_lib.core.LibConstants

class ModGeoArmorModel<T : GeoAnimatable>(
	modelPath: ResourceLocation,
	textureName: ResourceLocation,
	animationsName: ResourceLocation
) : GeoModelExpand<T>(
	modelPath(getPath(modelPath)), texturePath(getPath(textureName)), animationsPath(getPath(animationsName))
) {
	constructor(pathName: ResourceLocation) : this(pathName, pathName, pathName)

	override val defaultModelResource: ResourceLocation =
		modelPath(LibConstants.modRl("armor/default.geo.json"))

	companion object {
		@JvmStatic
		fun getPath(name: ResourceLocation): ResourceLocation {
			return name.withPrefix("armor/")
		}
	}
}
